src/FakeItEasy/ExpressionExtensions.cs

```diff
@@ -123,14 +124,16 @@ private static bool TryFastEvaluate(Expression expression, out object result)
 
                 case ExpressionType.NewArrayInit:
                     var newArrayExpression = (NewArrayExpression)expression;
-                    object[] arrayItems = new object[newArrayExpression.Expressions.Count];
+                    Array arrayItems = (Array)Activator.CreateInstance(newArrayExpression.Type, newArrayExpression.Expressions.Count);

```

Comment:
Wouldn't `Array.CreateInstance` be faster?

Reply:
Maybe! I did not think about it. Will do.

Reply:
Oh, I'll have to "de-array" the type. Still worth it?

Reply:
I did a quick benchmark, it seems to be about 13 times as fast